packages/ddl-shim/src/upgrade.ts

```diff
@@ -702,13 +703,18 @@ class DDLUpgrade {
         return retVal;
     }
 
-    writeProperties(): DDL2.IWidgetProperties {
-        return {};
+    writeProperties(): DDL2.IProperties | undefined {
+        //  TODO:  Upgrade v1.x.x dermatologies  ---
+        return undefined;
```

Comment:
What will be the arguments for this `writeProperties`? IOW, will it take in one widget's properties and output v2 of that one widget's properties? Or will it take in a v1 marshaller serialization and output a v2 marshaller serialization?

Reply:
It will take the v1 dermatology "layout" and upgrade it into v2 format.